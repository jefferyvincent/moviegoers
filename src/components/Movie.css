.movie {
    display: block;
    max-width: 246px;
    margin: 0.5rem;
    border: 1px solid grey;
    text-decoration: none;
}

.movie-poster {
    max-width: 100%;
    min-height: 400px;
}

.movie-text {
    color: grey;
    margin: 1rem;
    text-transform: uppercase;
    font-weight: bold;
    font-size: 0.75em;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
}