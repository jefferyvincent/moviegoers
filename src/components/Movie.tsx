import React from 'react';
import { Link } from "react-router-dom";

import './Movie.css';
import mi from '../img/missing_img.png';

interface MovieProps {
    id: number,
    img: string,
    title: string
}

const Movie: React.FC<MovieProps> = ({id, img, title}) => {
    const url = "/movie/" + id;
    let src = "https://image.tmdb.org/t/p/w500/"+img;

    if (img === null) {
        src = mi;
    }
    
    return (
        <Link to={url} className="movie" title={title}>
            <img className="movie-poster" src={src} alt={title} />
            <div className="movie-text">{title}</div>
        </Link>
    )
}

export default Movie;