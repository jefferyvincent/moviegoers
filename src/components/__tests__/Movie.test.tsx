import React from 'react';
import { shallow } from 'enzyme';
import {render } from 'react-testing-library';
import {HashRouter as Router} from "react-router-dom";
import Movie from '../Movie';

describe('<Movie/>', () => {
  let wrapper:any;
  beforeEach(() => { wrapper = shallow(<Movie id={1} img="test.png" title="test" />); });

  it("renders", () => {
    const { asFragment } = render(<Router><Movie id={1} img="test.png" title="test" /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('renders an movie title', () => {
    expect(wrapper.find('div.movie-text').length).toEqual(1);
  });
  it('renders an movie image', () => {
    expect(wrapper.find('img.movie-poster').length).toEqual(1);
  });
});