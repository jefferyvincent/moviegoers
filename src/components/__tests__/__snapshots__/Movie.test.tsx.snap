// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movie/> renders 1`] = `
<DocumentFragment>
  <a
    class="movie"
    href="#/movie/1"
    title="test"
  >
    <img
      alt="test"
      class="movie-poster"
      src="https://image.tmdb.org/t/p/w500/test.png"
    />
    <div
      class="movie-text"
    >
      test
    </div>
  </a>
</DocumentFragment>
`;
